### PR TITLE
refactor:  Separate type signature conversion into its own module

### PR DESCRIPTION
**Pull request by EpsilonPrime**
Moves proto type (not the in-memory type Type) to type signature conversion routine into its own module.  The reverse conversion will join this routine in an upcoming PR.

### CODE REVIEW

File: src/substrait/textplan/TypeConversion.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+namespace io::substrait::textplan {
+
+namespace {
+
+std::string kNullSignature = "?";
+
+} // namespace
+
+// Converts a Type proto directly into its text signature form.
+std::string typeToText(const ::substrait::proto::Type& type) {
```
Comment (chaojun-zhang) [inline src/substrait/textplan/TypeConversion.cpp:16]: how about refactor this method as  `std::share_ptr<ParameterType> fromSubstraitType(const ::substrait::proto::Type& type)` and move into header `Type.h` , and then we can get the signature via  ParameterType::signature() ?

Comment (EpsilonPrime) [reply src/substrait/textplan/TypeConversion.cpp:16]: I considered including the methods there but I was concerned about adding the protobufs as a dependency.  If this isn't an issue I'm up for moving this there.  @chaojun-zhang

File: src/substrait/textplan/TypeConversion.cpp [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  switch (type.kind_case()) {
+    case ::substrait::proto::Type::kBool:
+      if (type.bool_().nullability() ==
+          ::substrait::proto::Type::NULLABILITY_NULLABLE) {
+        return TypeTraits<TypeKind::kBool>::kSignature + kNullSignature;
```
Comment (chaojun-zhang) [inline src/substrait/textplan/TypeConversion.cpp:21]: why use  kSignature as text string  for bool type here and kTypeString for other types?

**Comment by EpsilonPrime** [comment]
I have a PR in progress that will add fromProto to the Substrait type module.  Then a followup PR to that will update PlanPrinterVisitor.cpp (the current location of this code) to use that functionality.

**Pull request closed without merging**